src/library_manager/lib_manager.c

```diff
@@ -47,6 +47,42 @@ struct lib_manager_dma_ext {
 
 static struct ext_library loader_ext_lib;
 
+struct lib_manager_mod_ctx {
+	struct sof_man_fw_desc *desc;
+	size_t segment_size[3];
```

Comment:
Minor: The magic 3 sticks out a but, some SOF_MAN_SEGMENT_COUNT ?

Reply:
@kv2019i well, here's how they're defined https://github.com/thesofproject/sof/blob/d324af52bd3ceaf502afa15bef12023af595491e/tools/rimage/src/include/rimage/sof/user/manifest.h#L47-L51 and it is the same as https://github.com/thesofproject/sof/blob/d324af52bd3ceaf502afa15bef12023af595491e/tools/rimage/src/include/rimage/sof/user/manifest.h#L103 so, yes, we could add a separate commit to define them and use them in both those structures, but that would modify manifest.h. Even though it wouldn't *actually* change anything, still that probably would be considered an ABI change? Or now?